pkg/config/types/v1alpha1/v1alpha1_configurator.go

```diff
@@ -302,6 +311,147 @@ func (e *EtcdConfig) ExtraArgs() map[string]string {
 	return e.EtcdExtraArgs
 }
 
+// Mirrors implements the Registries interface.
+func (r *RegistriesConfig) Mirrors() map[string]machine.RegistryMirrorConfig {
+	return r.RegistryMirrors
+}
+
+// Config implements the Registries interface.
+func (r *RegistriesConfig) Config() map[string]machine.RegistryConfig {
+	return r.RegistryConfig
+}
+
+// ExtraFiles implements the Registries interface.
+//
+//nolint: gocyclo
+func (r *RegistriesConfig) ExtraFiles() ([]machine.File, error) {
+	// config structures to generate TOML containerd CRI plugin config
+	type mirror struct {
```

Comment:
need to take it out of this file, seems to confuse docgen, and it doesn't look clean